tests/ops/op_math/test_controlled_decompositions.py

```diff
@@ -0,0 +1,127 @@
+"""
+from functools import reduce
+import pytest
+
+import numpy as np
+import pennylane as qml
+from pennylane.ops import ctrl_decomp_zyz
+from pennylane.wires import Wires
+
+
+class TestControlledDecompositionZYZ:
+    """tests for qml.ops.ctrl_decomp_zyz"""
+
+    def test_invalid_op_error(self):
+        """Tests that an error is raised when an invalid operation is passed"""
+        with pytest.raises(
+            ValueError, match="The target operation must be a single-qubit operation"
+        ):
+            _ = ctrl_decomp_zyz(qml.CNOT([0, 1]), [2])
+
+    su2_ops = [
+        qml.RX(0.123, wires=0),
+        qml.RY(0.123, wires=0),
+        qml.RZ(0.123, wires=0),
+        qml.Rot(0.123, 0.456, 0.789, wires=0),
+    ]
+
+    unitary_ops = [
+        qml.Hadamard(0),
+        qml.PauliZ(0),
+        qml.S(0),
+        qml.PhaseShift(1.5, wires=0),
+        qml.QubitUnitary(
+            np.array(
+                [
+                    [-0.28829348 - 0.78829734j, 0.30364367 + 0.45085995j],
+                    [0.53396245 - 0.10177564j, 0.76279558 - 0.35024096j],
+                ]
+            ),
+            wires=0,
+        ),
+        qml.DiagonalQubitUnitary(np.array([1, -1]), wires=0),
+    ]
+
+    @pytest.mark.parametrize("op", su2_ops + unitary_ops)
+    @pytest.mark.parametrize("control_wires", ([1], [1, 2], [1, 2, 3]))
+    def test_decomposition_circuit(self, op, control_wires, tol):
+        """Tests that the controlled decomposition of a single-qubit operation
+        behaves as expected in a quantum circuit"""
+        decomps = ctrl_decomp_zyz(op, Wires(control_wires))
+        expected_op = qml.ctrl(op, control_wires)
+
+        dev = qml.device("default.qubit", wires=4)
+
+        @qml.qnode(dev)
+        def decomp_circuit():
+            for i in range(4):
+                qml.Hadamard(i)
+            for decomp_op in decomps:
+                qml.apply(decomp_op)
+            return qml.probs()
+
+        @qml.qnode(dev)
+        def expected_circuit():
+            for i in range(4):
+                qml.Hadamard(i)
+            qml.apply(expected_op)
+            return qml.probs()
+
+        res = decomp_circuit()
+        expected = expected_circuit()
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("op", su2_ops)
+    @pytest.mark.parametrize("control_wires", ([1], [1, 2], [1, 2, 3]))
+    def test_decomposition_matrix(self, op, control_wires, tol):
+        """Tests that the matrix representation of the controlled ZYZ decomposition
+        of a single-qubit operation is correct"""
+        decomps = ctrl_decomp_zyz(op, Wires(control_wires))
+        expected_op = qml.ctrl(op, control_wires)
+        decomp_mats = [
+            np.kron(np.eye(2 ** len(control_wires)), decomp_op.matrix())
+            if not isinstance(decomp_op, qml.MultiControlledX)
+            else decomp_op.matrix()
+            for decomp_op in decomps
+        ]
```

Comment:
You should be able to just do `qml.matrix(ctrl_decomp_zyz)(op, control_wires)`. `ctrl_decomp_zyz` should behave like any other quantum function.